### PR TITLE
Some minor fixes

### PR DESCRIPTION
**Pull request by XxRenderxX**
**Changes proposed:**

-  Fixed the Typo in the Stealth Prevention Spell Code
-  Added a fixed new warriors/death knights not having their stances active at first login (All credits to TrinityCore/TrinityCore@6ff3d37)

**Tests performed:** (Does it build, tested in-game, etc)

Tested and stance fix does work. (Could use a confirmation for it though) 

**Known issues and TODO list:**

None so far.

### CODE REVIEW

**Comment by XxRenderxX** [comment]
Finally fixed the typo thanks, Falzard

File: src/game/Handlers/CharacterHandler.cpp [hunk 1]
```diff
@@ -1071,9 +1071,26 @@ void WorldSession::HandlePlayerLoginFromDB(LoginQueryHolder* holder)
         pCurrChar->SendTalentsInfoData(false);              // original talents send already in to SendInitialPacketsBeforeAddToMap, resend reset state
         SendNotification(LANG_RESET_TALENTS);
     }
-
-    if (pCurrChar->HasAtLoginFlag(AT_LOGIN_FIRST))
-        pCurrChar->RemoveAtLoginFlag(AT_LOGIN_FIRST);
+    
+    // Fix done by mthsena;
+    bool firstLogin = pCurrChar->HasAtLoginFlag(AT_LOGIN_FIRST);
+      if (firstLogin)
```
Comment (ShinDarth) [inline src/game/Handlers/CharacterHandler.cpp]: please fix the indentation

File: src/game/Handlers/CharacterHandler.cpp [hunk 2]
```diff
@@ -1071,9 +1071,26 @@ void WorldSession::HandlePlayerLoginFromDB(LoginQueryHolder* holder)
         pCurrChar->SendTalentsInfoData(false);              // original talents send already in to SendInitialPacketsBeforeAddToMap, resend reset state
         SendNotification(LANG_RESET_TALENTS);
     }
-
-    if (pCurrChar->HasAtLoginFlag(AT_LOGIN_FIRST))
-        pCurrChar->RemoveAtLoginFlag(AT_LOGIN_FIRST);
+    
+    // Fix done by mthsena;
```
Comment (ShinDarth) [inline src/game/Handlers/CharacterHandler.cpp]: we do not add credits using comments in the code, the credits to the author are added automatically thanks to the versioning system

File: src/game/Handlers/CharacterHandler.cpp [hunk 3]
```diff
@@ -1071,9 +1071,26 @@ void WorldSession::HandlePlayerLoginFromDB(LoginQueryHolder* holder)
...
+          pCurrChar->RemoveAtLoginFlag(AT_LOGIN_FIRST);
+  
+          // Activate [Battle Stance] and [Blood Presence] at first login.
+          switch (pCurrChar->getClass())
+          {
+          case CLASS_WARRIOR:
+              pCurrChar->CastSpell(pCurrChar, 2457, true);
+              break;
+          case CLASS_DEATH_KNIGHT:
+              pCurrChar->CastSpell(pCurrChar, 48266, true);
```
Comment (ShinDarth) [inline src/game/Handlers/CharacterHandler.cpp]: please wrap those numbers into constants

**Comment by ShinDarth** [comment]
thanks for the PR, please check my comments and update your code

**Comment by talamortis** [comment]
tested and no change for me players didnt start with stances already enabled

**Comment by ShinDarth** [comment]
@talamortis @XxRenderxX can you please discuss it?

**Comment by XxRenderxX** [comment]
Odd, it's working great for me for both Warriors and Death Knights.
Here's a short video for it: https://youtu.be/djktY5zfJH4

Try applying to a fresh AzerothCore Compile.

**Comment by Yehonal** [comment]
It's an horrible hack fix, please at least....use a ScriptMgr hook to place this code out of code since we should avoid having specific WoW spells handled inside functions such as HandlePlayerLoginFromDB

there's an hook available on ScriptMgr for first login

**Comment by StygianTheBest** [comment]
@Yehonal 

> there's an hook available on ScriptMgr for first login

I created a module that added items/spells/etc. on first login, but it was throwing errors.  I followed Rochet's addition to add Trinity and it's been working fine on my server. I can't remember if I was using ScriptMgr or not. 

https://github.com/TrinityCore/TrinityCore/commit/dae3ae069651ed3f35f975091af59a33181ef0e0

**Comment by Yehonal** [comment]
better alternative: https://github.com/azerothcore/azerothcore-wotlk/pull/562

**Pull request closed without merging**